src/test/kotlin/org/rust/ide/annotator/RsEdition2018KeywordsAnnotatorTest.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import org.rust.MockEdition
+import org.rust.cargo.project.workspace.CargoWorkspace
+
+class RsEdition2018KeywordsAnnotatorTest : RsAnnotationTestBase() {
+
+    fun `test edition 2018 keywords in edition 2015`() = checkErrors("""
+        fn main() {
+            let async = 1;
+            let await = 2;
+            let try = 3;
+            let x = async;
+            let y = await;
+            let z = try;
+        }
+    """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test edition 2018 keywords in edition 2018`() = checkErrors("""
+        fn main() {
+            let <error descr="`async` is reserved keyword in Edition 2018">async</error> = 1;
+            let <error descr="`await` is reserved keyword in Edition 2018">await</error> = 2;
+            let <error descr="`try` is reserved keyword in Edition 2018">try</error> = 3;
+            let x = <error descr="`async` is reserved keyword in Edition 2018">async</error>;
+            let y = <error descr="`await` is reserved keyword in Edition 2018">await</error>;
+            let z = <error descr="`try` is reserved keyword in Edition 2018">try</error>;
+        }
+    """)
+
+    // We should report an error here
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test reserved keywords in macro names in edition 2018`() = checkErrors("""
+        fn main() {
+            let x = async!();
+            let y = await!();
+            let z = try!(42);
+        }
+    """)
+
+    fun `test async in edition 2015`() = checkErrors("""
+        <error descr="This feature is only available in Edition 2018">async</error> fn foo() {}
+
+        fn main() {
+            <error descr="This feature is only available in Edition 2018">async</error> { 42 };
+            <error descr="This feature is only available in Edition 2018">async</error> || { 42 };
+            <error descr="This feature is only available in Edition 2018">async</error> move || { 42 };
+        }
+    """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test async in edition 2018`() = checkErrors("""
+        async fn foo() {}
+
+        fn main() {
+            async { 42 };
+            async || { 42 };
+            async move || { 42 };
+        }
```

Comment:
```
async move {}
```

should be supported.

https://github.com/rust-lang/rfcs/blob/master/text/2394-async_await.md#initialization-pattern